src/models/events.rs

```diff
@@ -0,0 +1,294 @@
+pub mod payload;
+
+use self::payload::{CreateEventPayload, EventPayload, PushEventPayload};
+use chrono::{DateTime, Utc};
+use reqwest::Url;
+use serde::{de::Error, Deserialize, Serialize};
+
+/// A GitHub event.
+#[derive(Debug, Clone, PartialEq, Serialize)]
+#[non_exhaustive]
+pub struct Event {
+    pub id: String,
+    #[serde(rename = "type")]
+    pub typ: EventType,
+    pub actor: Actor,
+    pub repo: Repository,
+    pub public: bool,
+    pub created_at: DateTime<Utc>,
+    pub payload: Option<EventPayload>,
+    pub org: Option<Org>,
+}
+
+/// The type of an event.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[non_exhaustive]
+pub enum EventType {
```

Comment:
Could you include a `#[serde(rename_all)]` attribute here, just so it's clear what their casing will be.